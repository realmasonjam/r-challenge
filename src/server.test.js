const app = require('./server')
const supertest = require('supertest')
const request = supertest(app)

it('gets the list of the properties when there is no property in DB', async done => {
  const response = await request.get('/')
  expect(response.status).toBe(200)
  expect(response.body).toEqual([])
  done()
})

it('Posts a property and get the same property', async done => {
  const postResponse = await request.post('/').send({
    "description": "Private Sale",
    "address": "17 Carroll Street, Richmond, Vic 3121",
    "price": 1500000,
    "suburb": "Richmond"
  })
  expect(postResponse.status).toBe(200)
  const getResponse = await request.get('/')
  expect(getResponse.status).toBe(200)
  expect(getResponse.body).toMatchSnapshot()
  done()
})

it('Adds 3 different properties and get all of them', async done => {
  const postResponse = await request.post('/').send(
    {
      "price": 4000000,
      "description": "Auction",
      "address": "1 Williams Road, Prahran, Vic 3181",
      "suburb": "Prahran"
    })
  expect(postResponse.status).toBe(200)
  await request.post('/').send({
      "price": 700000,
      "suburb": "Prahran",
      "address": "1/78-80 Commercial Road, Prahran, Vic 3181",
      "description": "Auction",
    })
  await request.post('/').send({
      "address": "21 St Edmonds Road, Prahran, Vic 3181",
      "description": "Private Sale",
      "suburb": "Prahran",
      "price": 600000,
    })
  const getResponse = await request.get('/?suburb=Prahran')
  expect(getResponse.status).toBe(200)
  expect(getResponse.body).toMatchSnapshot()
  done()
})


