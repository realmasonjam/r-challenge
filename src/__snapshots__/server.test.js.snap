// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adds 3 different properties and get all of them 1`] = `
Array [
  Object {
    "address": "1 Williams Road, Prahran, Vic 3181",
    "description": "Auction",
    "price": 4000000,
    "priceRange": "Greater than average",
    "suburb": "Prahran",
    "suburbAveragePrice": 1766666.6666666667,
  },
  Object {
    "address": "1/78-80 Commercial Road, Prahran, Vic 3181",
    "description": "Auction",
    "price": 700000,
    "priceRange": "Lower than average",
    "suburb": "Prahran",
    "suburbAveragePrice": 1766666.6666666667,
  },
  Object {
    "address": "21 St Edmonds Road, Prahran, Vic 3181",
    "description": "Private Sale",
    "price": 600000,
    "priceRange": "Lower than average",
    "suburb": "Prahran",
    "suburbAveragePrice": 1766666.6666666667,
  },
]
`;

exports[`Posts a property and get the same property 1`] = `
Array [
  Object {
    "address": "17 Carroll Street, Richmond, Vic 3121",
    "description": "Private Sale",
    "price": 1500000,
    "priceRange": "Equal to average",
    "suburb": "Richmond",
    "suburbAveragePrice": 1500000,
  },
]
`;
